README.md

```diff
@@ -220,6 +220,159 @@ C:\Users\bion\Desktop>dumpbin /IMPORTS .\example.exe | findstr msvcp
+
+### CTest
+
+When running the tests via CTest each of the testsuites is considered to be a single test. If any single test in a
+testsuite fails, CTest will report the test which represents that testsuite as failed.
+
+Example:
+```
+67% tests passed, 1 tests failed out of 3
+
+Total Test time (real) = 2441.55 sec
+
+The following tests FAILED:
+      1 - libcxx (Failed)
+```
+
+CTest will output everything that was sent to stderr for each of the failed testsuites, which can be used to identify
+which individual test within the testsuite failed. It can sometimes be helpful to run CTest with the `-V` option in
+order to see the stdout of the tests.
+
+### llvm-lit
+
+When running the tests directly via the generated `llvm-lit.py` script the result of each test will be printed. The
+format of each result is `{Result Code}: {Testsuite Name} :: {Test Name}:{Configuration Number}`.
+
+Example:
+```
+-- Testing: 28 tests, 12 workers --
+PASS: tr1 :: tests/cwchar1:01 (1 of 28)
+PASS: tr1 :: tests/cwchar1:11 (2 of 28)
+PASS: tr1 :: tests/cwchar1:02 (3 of 28)
+PASS: tr1 :: tests/cwchar1:03 (4 of 28)
+PASS: tr1 :: tests/cwchar1:00 (5 of 28)
+PASS: tr1 :: tests/cwchar1:04 (6 of 28)
+PASS: tr1 :: tests/cwchar1:05 (7 of 28)
+PASS: tr1 :: tests/cwchar1:09 (8 of 28)
+PASS: tr1 :: tests/cwchar1:06 (9 of 28)
+UNSUPPORTED: tr1 :: tests/cwchar1:20 (10 of 28)
+UNSUPPORTED: tr1 :: tests/cwchar1:21 (11 of 28)
+UNSUPPORTED: tr1 :: tests/cwchar1:22 (12 of 28)
+UNSUPPORTED: tr1 :: tests/cwchar1:23 (13 of 28)
+UNSUPPORTED: tr1 :: tests/cwchar1:24 (14 of 28)
+PASS: tr1 :: tests/cwchar1:07 (15 of 28)
+PASS: tr1 :: tests/cwchar1:08 (16 of 28)
+PASS: tr1 :: tests/cwchar1:10 (17 of 28)
+PASS: tr1 :: tests/cwchar1:16 (18 of 28)
+PASS: tr1 :: tests/cwchar1:17 (19 of 28)
+PASS: tr1 :: tests/cwchar1:14 (20 of 28)
+PASS: tr1 :: tests/cwchar1:12 (21 of 28)
+PASS: tr1 :: tests/cwchar1:13 (22 of 28)
+PASS: tr1 :: tests/cwchar1:19 (23 of 28)
+PASS: tr1 :: tests/cwchar1:18 (24 of 28)
+PASS: tr1 :: tests/cwchar1:15 (25 of 28)
+PASS: tr1 :: tests/cwchar1:25 (26 of 28)
+PASS: tr1 :: tests/cwchar1:26 (27 of 28)
+PASS: tr1 :: tests/cwchar1:27 (28 of 28)
+
+Testing Time: 3.96s
+  Expected Passes    : 23
+  Unsupported Tests  : 5
+```
+
+In the above example we see that 23 tests succeeded and 5 were unsupported.
+
+### Result Code Values
+
+Our tests use the standard [lit result codes][], and a non-standard result code: `SKIP`. For our tests, only the
+`PASS`, `XFAIL`, `XPASS`, `FAIL`, and `UNSUPPORTED` standard result codes are relevant.
+
+The `PASS` and `FAIL` result codes are self-explanatory. We want our tests to `PASS` and not `FAIL`.
+
+The `XPASS` and `XFAIL` result codes are less obvious. `XPASS` is actually a failure result and indicates that we
+expected a test to fail but it passed. `XFAIL` is a successful result and indicates that we expected the test to fail
+and it did. Typically an `XPASS` result means that the `expected_results.txt` file for the testsuite needs to be

```

Comment:
I gues you want a negation here `and it did *not`

Reply:
No I think what is written is correct. `XFAIL` means that we expected the test to fail and it did. `XPASS` means we expected the test to fail but it passed.

I know, it's a little confusing.

Reply:
The full sentence is:

>`XFAIL` is a successful result and indicates that we expected the test to fail
and it did.

So we expected the test to fail and it did?